### ISSUE TITLE
"Post from URL" requests unnecessary permissions

### ISSUE THREAD

**Issue by mahkoh** [ccd0/4chan-x#1027]
Whenever I want to post an image from a URL, 4chan-X requests additional permissions. Then I click "Deny" and everything just works anyway. So why does it request the permissions in the first place?

**Comment by mahkoh** [comment]
The permission is "Read and change all your data on the websites you visit".

**Comment by ccd0** [comment]
What was the URL?

**Comment by mahkoh** [comment]
https://i.imgur.com/JHsRkBj.jpg for example. I'm using https on 4chan if that makes a difference.

**Comment by ccd0** [comment]
It seems Imgur sends the `Access-Control-Allow-Origin: *` header which allows the request to work without extra permissions. For most sites you would need the permission.

**Comment by mahkoh** [comment]
Can you give me an example? I've never had a problem uploading images with this feature and it has never asked for permissions until a few days ago.

**Comment by ccd0** [comment]
Here's one:
http://i.telegraph.co.uk/multimedia/archive/02830/cat_2830677b.jpg

It didn't ask for permissions until recently because it used to ask for the permissions on install.

**Comment by mahkoh** [comment]
It works for me even if I deny the permission: https://boards.4chan.org/qa/thread/594136#p597178

Note that I haven't reinstalled 4chan-x recently. So maybe it's actually using another permission that I already gave it before for this feature? The dialog does say that 4chan-x is requesting **additional** permissions.

PS: So I just checked the extension dialog and it says that 4chan-x already has this permission:

![2016-07-22-000006_366x348_scrot](https://cloud.githubusercontent.com/assets/1882250/17040346/6279b20e-4f9f-11e6-9ee3-caef29471a1f.png)

Don't know if this is a bug in 4chan-x or chrome but I'll just give it the permission again.
